src/6model/reprs/P6bigint.c

```diff
@@ -4,30 +4,27 @@
    #define MIN(x,y) ((x)<(y)?(x):(y))
 #endif
 
-/* A forced 64-bit version of mp_get_long, since on some platforms long is
- * not all that long. */
-static MVMuint64 mp_get_int64(MVMThreadContext *tc, mp_int * a) {
-    int i, bits;
+/* Handles both int and uint cases. */
+static MVMuint64 mp_get_int64(MVMThreadContext *tc, mp_int * a, int is_signed) {
```

Comment:
I did test that, but I believe the problem was because of the lack of some other *_u ops. Some things end up calling the *_int version when they really mean/want the *_uint version, and then you run into the exact same problem that they can have different numbers of bits allowed.